src/construction.cpp

```diff
@@ -1241,6 +1243,30 @@ bool construct::check_support( const tripoint_bub_ms &p )
     return num_supports >= 2;
 }
 
+bool construct::check_support_below( const tripoint_bub_ms &p )
+{
+    bool blocking_creature = g->get_creature_if( [&]( const Creature & creature ) {
+        return creature.pos() == p.raw();
+    } ) != nullptr;
+
+    map &here = get_map();
+    if( !( here.passable( p.raw() ) || here.has_flag( ter_furn_flag::TFLAG_LIQUID, p ) ||
+           here.ter( p ) == t_open_air ) ||
+        blocking_creature || here.has_furn( p ) || !( here.tr_at( p ).is_null() ||
+                here.tr_at( p ).id == tr_ledge ) ||
+        !here.i_at( p ).empty() || here.veh_at( p ) ) {
```

Comment:
There's a bunch of convoluted conditions you have here. Maybe add some comments on what are you trying to check?

Reply:
Added comment in commit.